trustymail/trustymail.py

```diff
@@ -165,71 +166,147 @@ def starttls_scan(domain, smtp_timeout, smtp_localhost, smtp_ports, smtp_cache):
-                # (64.69.57.18) since it (1) has a valid PTR record and (2) is
-                # not listed by anyone as a valid mail server.
-                #
-                # I'm actually temporarily using an IP that
-                # virginia.edu resolves to until we resolve why Google
-                # DNS does not return the same PTR records as the CAL
-                # DNS does for 64.69.57.18.
-                query = spf.query("128.143.22.36", "email_wizard@" + domain.domain_name, domain.domain_name, strict=2)
-                response = query.check()
-            except spf.AmbiguityWarning as error:
-                logging.debug("\t" + error.msg)
-                domain.syntax_errors.append(error.msg)
-                continue
+                record_to_return = record_text
+    except (dns.resolver.NoNameservers, dns.resolver.NoAnswer, dns.exception.Timeout, dns.resolver.NXDOMAIN) as error:
+        handle_error('[SPF]', domain, error)
 
-            if response[0] == 'temperror':
-                logging.debug(response[2])
-            elif response[0] == 'permerror':
-                logging.debug("\t" + response[2])
-                domain.syntax_errors.append(response[2])
-            elif response[0] == 'ambiguous':
-                logging.debug("\t" + response[2])
-                domain.syntax_errors.append(response[2])
-            elif response[0] == result:
-                # Everything checks out the SPF syntax seems valid.
-                domain.valid_spf = True
-                continue
-            else:
-                domain.valid_spf = False
-                logging.debug("\tResult Differs: Expected [{0}] - Actual [{1}]".format(result, response[0]))
-                domain.errors.append("Result Differs: Expected [{0}] - Actual [{1}]".format(result, response[0]))
+    return record_to_return
 
-    except (dns.resolver.NoNameservers, dns.resolver.NoAnswer, dns.exception.Timeout, dns.resolver.NXDOMAIN) as error:
-        handle_error("[SPF]", domain, error)
+
+def spf_scan(resolver, domain):
+    """
+    Scan a domain to see if it supports SPF.  If the domain has an SPF
+    record, verify that it properly rejects mail sent from an IP known
+    to be disallowed.
+
+    Parameters
+    ----------
+    resolver : dns.resolver.Resolver
+        The Resolver object to use for DNS queries.
+        
+    domain : trustymail.Domain 
+        The Domain object being scanned for SPF support.  Any errors
+        will be logged to this object.
+    """
+    # If an SPF record exists, record the raw SPF record text in the
+    # Domain object
+    record_text_not_following_redirect = get_spf_record_text(resolver, domain.domain_name, domain)
+    if record_text_not_following_redirect:
+        domain.spf.append(record_text_not_following_redirect)
+    
+    record_text_following_redirect = get_spf_record_text(resolver, domain.domain_name, domain, True)
```

Comment:
`get_spf_record_text` is called with `follow_redirect=True` here because we want to follow the redirects until we get an SPF record that we can use to determine the expected behavior when we run `spf.check()` inside of the `check_spf_record` function on line 198.